src/cljs/rems/application.cljs

```diff
@@ -773,7 +773,8 @@
                        (map dynamic-event->event (:dynamic-events app)))
         applicant-attributes (:applicant-attributes application)
         messages (remove nil?
-                         [(disabled-items-warning (:catalogue-items application)) ; NB: eval this here so we get nil or a warning
+                         [(when (editable? application)
```

Comment:
This check should be moved inside `disabled-items-warning`. However, there are also other needs as discussed so let's specify this task a bit better together.